pkg/scheduler/util/features.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "k8s.io/component-base/featuregate"
+
+// NOTE: constants in this file are copied from pkg/features package to avoid external scheduler dependency
```

Comment:
scoping feature gates per component seems like a natural progression.
during this process there is the potential for default state of FGs (enabled for >= beta) to drift between the core (pkg/features) and the component that forked the FG. interesting how do you plan to handle that?

@kubernetes/sig-architecture-pr-reviews

Reply:
I am not sure what is the common practices for other components. I think the worst situation may be synchronizing it before each release?

Reply:
from my perspective, FGs that target a single component should only be known by this component instead of residing in a shared pool.

on the other hand feature gates that apply to multiple components may have a separate lifecycle. so SomeFeatureGate can be Alpha in the ComponentA and Beta at ComponentB. if SomeFeatureGate graduates to Beta in both that's great, but maybe the overall feature should be considered Alpha (lowest version between ComponentA-B).

Reply:
I don't think that replicating these definitions is a good idea. All but two of these apply to multiple components and their definitions will drift.

>  if SomeFeatureGate graduates to Beta in both that's great. but maybe the overall feature should be considered the lowest version between ComponentA-B.

that's really confusing for a cluster admin... if there are different aspects that need enabling differently per component, those are probably distinct gates (xref EndpointSlice and EndpointSliceProxying)

An alternative to the scheduler using a global feature map would be to inject the feature states depended on by the scheduler in the scheduler.New() call, so only the cmd/kube-scheduler/app package would depend on the feature gates

Reply:
Only `BalanceAttachedNodeVolumes` and `ResourceLimitsPriorityFunction`(one IT also depends on this) FGs are only used in scheduler.

/cc @Huang-Wei @damemi @ahg-g @alculquicondor    WDYT about handling FG in staging scheduler?

Reply:
I also think this is a bad idea. Feature Gates have always been k8s-wide. So it's not a pattern that can be changed lightly.
Certainly, there are features that involve a number of components and features that don't. But we still release all k8s binaries together, so it doesn't make much sense to treat them separately at this point.
Maybe FG should be part of k8s.io/component-base instead.

Reply:
I have created an issue to track this at #90136.